src/mlpack/methods/ann/init_rules/xavier_init.hpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file xavier_init.hpp
+ * @author kris singh
+ *
+ * Intialization rule for the neural networks. This simple initialization is
+ * performed by assigning a weights from a uniform distribution between 2/fan_in + fan_out.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_INIT_RULES_XAVIER_INIT_HPP
+#define MLPACK_METHODS_ANN_INIT_RULES_XAVIER_INIT_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+#include "gaussian_init.hpp"
+
+using namespace mlpack::math;
+
+ 
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+class XavierNormal
+{};
+class XavierUniform
+{};
```

Comment:
There's so little overlap in the code here that I think maybe it is better to simply have a XavierUniformInit and XavierNormalInit class. (The same applies for HeInit.)